src/mgmt/srv.c

```diff
@@ -300,6 +303,106 @@ process_join_request(Drpc__Call *drpc_req, Mgmt__JoinResp *resp)
 		resp->status = MGMT__DAOS_REQUEST_STATUS__ERR_UNKNOWN;
 }
 
+static void
+process_create_pool_request(Drpc__Call *drpc_req, Mgmt__CreatePoolResp *resp)
+{
+	Mgmt__CreatePoolReq	*pb_req = NULL;
+	d_rank_list_t		*targets = NULL;
+	d_rank_t		ranks[max_svc_nreplicas];
+	d_rank_list_t		svc = {};
+	d_rank_list_t		*svc_p = &svc;
+	uuid_t			pool_uuid;
+	int			buflen = 16;
+	int			len;
+	int			i;
+	int			rc = 0;
+	char			id[DAOS_UUID_STR_SIZE];
+	char			buf[buflen];
+	char			*svc_str = buf;
+	char			*extra = NULL;
+
+	memset(ranks, 0, sizeof(ranks));
+	svc_p->rl_ranks = ranks;
+
+	/* response status is populated with SUCCESS on init */
+	mgmt__create_pool_resp__init(resp);
+
+	/* Unpack the daos request from the drpc call body */
+	pb_req = mgmt__create_pool_req__unpack(
+		NULL, drpc_req->body.len, drpc_req->body.data);
+
+	if (pb_req == NULL) {
+		resp->status = MGMT__DAOS_REQUEST_STATUS__ERR_UNKNOWN;
+		D_ERROR("Failed to extract request\n");
+
+		return;
+	}
+
+	/* parse targets rank list */
+	if (strlen(pb_req->ranks) != 0) {
+		targets = daos_rank_list_parse(pb_req->ranks, ",");
+		if (targets == NULL) {
+			D_ERROR("failed to parse target ranks\n");
+			D_GOTO(out, rc = -DER_UNKNOWN);
```

Comment:
Do not add new D_GOTOs; use goto. Also, this function is void; use dRPC way to return errors.

Reply:
I was changing rc to indicate that we need to add failure into the response in "out"

Reply:
if I need to set rc to indicate that an error occurred so resp->status is set appropriately in "out", why is it not appropriate to use D_GOTO to set rc on the same line even if we are not returning rc?